File: String/longest_common_prefix.kt
package String


fun main() {

    var arr = arrayOf("flower", "flow" , "flight")
    longestCommonPrefix(arr)
}
fun longestCommonPrefix(strs: Array<String>): String? {
    if (strs.size == 0) return ""
    var prefix = strs[0]
    for (i in 1 until strs.size)
        while (strs[i].indexOf(prefix) != 0) {
        println(strs[i])
            // flower   flowe   flow    flo
        prefix = prefix.substring(0, prefix.length - 1)
//            println(prefix)
            if (prefix.isEmpty()) return ""
        }
    return prefix
}

File: Array/longest_common_prefix.py
from typing import List


'''
And above is what you are looking at here 

Input: strs = ["flower","flow","flight"]
Output: "fl"

the above give you what you are looking for here 
'''

def longestCommonPrefix(strs: List[str]) -> str:
    if len(strs) == 0:
        return ""
    prefix = strs[0]
    for i in range(1, len(strs)):
        '''
        Until we find it we will keep going at this 
        '''
        while strs[i].find(prefix) != 0:
            '''
            We basically keep cutting this down until we get to the prefix here fl 
            '''
            prefix = prefix[0: len(prefix) - 1]
            print("prefix is", prefix)
            if prefix == "":
                return ""
    return prefix
strs = ["flower","flow","flight"]
longestCommonPrefix(strs)